pkg/validation/attributes/attributes.go

```diff
@@ -0,0 +1,62 @@
+package attributes
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	apiAttributes "github.com/devfile/api/v2/pkg/attributes"
+)
+
+// ValidateGlobalAttribute validates the workspace template spec data for global attribute references
+func ValidateGlobalAttribute(workspaceTemplateSpec *v1alpha2.DevWorkspaceTemplateSpec) error {
```

Comment:
This function (and the subfunctions below) are named `Validate[...]` but in reality they appear to validate *and* replace attributes. We should make this functionality clear (and potentially have purely validation-focussed paths)

Reply:
Does `ValidateAndReplaceFor[...]` sound ok to you? I'll also update the func desc.

Reply:
Makes sense

Reply:
Addressed